hibernate-core/src/main/java/org/hibernate/engine/query/spi/QueryPlanCache.java

```diff
@@ -200,17 +200,26 @@ public FilterQueryPlan getFilterQueryPlan(
 			Map<String,Filter> enabledFilters) throws QueryException, MappingException {
 		final FilterQueryPlanKey key =  new FilterQueryPlanKey( filterString, collectionRole, shallow, enabledFilters );
 		FilterQueryPlan value = (FilterQueryPlan) queryPlanCache.get( key );
+		final StatisticsImplementor statistics = factory.getStatistics();
+		boolean stats = statistics.isStatisticsEnabled();
```

Comment:
the 'stats' seems too terse and might be renamed to 'isStatEnabled' or stuff like that? `final` keyword is desirable as well.